openedx/core/djangoapps/course_date_signals/handlers.py

```diff
@@ -53,26 +76,13 @@ def extract_dates_from_course(course):
             # Apply the same relative due date to all content inside a section,
             # unless that item already has a relative date set
             for _, section, weeks_to_complete in spaced_out_sections(course):
-                items = [section]
-                has_non_ora_scored_content = False
                 section_date_items = []
-                while items:
-                    next_item = items.pop()
-                    # TODO: This is pretty gross, and should maybe be configurable in the future,
-                    # especially if we find ourselves needing more exceptions.
-                    if next_item.graded:
-                        # TODO: Once studio can manually set relative dates,
-                        # we would need to manually check for them here
-                        section_date_items.append((next_item.location, {'due': weeks_to_complete}))
-                        has_non_ora_scored_content = (
-                            has_non_ora_scored_content or
-                            (next_item.has_score and next_item.category != 'openassessment')
-                        )
-
-                    items.extend(next_item.get_children())
-
-                if has_non_ora_scored_content:
-                    date_items.extend(section_date_items)
+                for subsection in section.get_children():
+                    section_date_items.extend(_gather_graded_items(subsection, weeks_to_complete))
+
+                if section_date_items and section.graded:
+                    date_items.append((section.location, weeks_to_complete))
+                date_items.extend(section_date_items)
```

Comment:
Why have this outside the if statement? Either the section_date_items are `[]` and there is nothing to add. Or they are not `[]`, but if the section isn't graded, this would still add the subsection dates and I'm not sure we want that. Just pondering out loud here.

Reply:
I just thought it looked cleaner this way. But I can be convinced it is too implicit or is confusing.

Reply:
My only concern would be for the case where `section_date_items != [] and section.graded = False` and we add subsection dates without adding in the section due date. I'm not sure this scenario can even happen, but might be good to be defensive about it

Reply:
I don't think that can happen (I believe `graded` is bubbled up the tree). But also, that is the behavior of the existing code. So it wouldn't be a regression at least.